codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -495,20 +518,22 @@ private void incrementPendingBytes(int numBytes) {
              * Note: this does not flush the {@link ChannelHandlerContext}.
              */
             int write(int allowedBytes) {
-                int before = payload.size();
-                needFlush |= payload.write(Math.max(0, allowedBytes));
-                int writtenBytes = before - payload.size();
                 try {
+                    pendingWriteQueue.remove();
```

Comment:
> error can still be called multiple times

My 2 cents...Unless we can come up with a cleaner approach I'd rather have it be called multiple times rather than requiring special code to prevent it (pop from queue, removing listeners, etc...). Maybe rename `error` to indicate this expected behavior...something like `tryError` or `tryFailure`?